### DISCUSSION TITLE
allow setting `syncTabs` after a store constructed

### DISCUSSION THREAD

**Discussion by CNSeniorious000** [joshnuss/svelte-local-storage-store#295] [Q&A] [answered]
Now we can't modify `syncTabs` or any other initialization options after creating a store by `persisted()`. Can we allow overriding this after creation?

**Comment by joshnuss** [comment]
Hi @CNSeniorious000,

One solution is to re-initialize the store with new options.

Let me know if that solves the issue

**Reply by CNSeniorious000** [reply]
Thanks @joshnuss! But what do you mean by _re-initialize the store_? Usually a store is a `const` instead of a `let`. And what if I've bound several subscribers on the initial persisted store?

What I want is like `spring` in `svelte/motion`, whose options can be modified after being created, and modifications will take effects immediately.

---

Let me explain my requirements further. I am using `svelte-persisted-store` to store some user preferences options in the application, but I the `syncTabs` option is a preference too. For now, I can only refresh the page to make that option take effect, which may not be a good UX.

**Reply by joshnuss** [reply, marked as answer]
I see,

Maybe there's another workaround

Like, what do you think about subscribing to the `storage` event manually and handling the syncing based on the user's preferences?

**Reply by CNSeniorious000** [reply]
Good idea!
